app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+
+module DynamicGraphs.GraphGenerator where
+
+import           Data.Graph.Inductive
+import           Data.GraphViz
+import           Data.GraphViz.Attributes.Complete
+import           Data.GraphViz.Types.Generalised   as G
+import           Data.GraphViz.Types.Monadic
+import           Data.Text.Lazy                    as L
+import           Data.Word
+import           DynamicGraphs.WriteRunDot
+
+ex3 :: G.DotGraph L.Text
+ex3 = digraph (Str "ex3") $ do
+
+    graphAttrs [RankDir FromLeft]
+
+    cluster (Str "G1") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 1]
+        node "A"                [textLabel "MAT237"]
+
+    cluster (Str "G2") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 1]
+        node "B"                [textLabel "STA261"]
+
+    cluster (Str "G3") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 1]
+        node "C"                [textLabel "MAT224"]
+
+    cluster (Str "G4") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 2]
+        node "D"                [textLabel "MAT137"]
+
+    cluster (Str "G5") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 2]
+        node "E"                [textLabel "MAT157"]
+
+    cluster (Str "G6") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 3]
+        node "F"                [textLabel "STA257"]
+
+    cluster (Str "G0") $ do
+        nodeAttrs               [shape Circle, Width 1, style filled, myColor 4]
+        node "ZZ"               [textLabel "STA302"]
+
+    ("A" :: Text)             --> ("ZZ" :: Text)
+    ("B" :: Text)              --> ("ZZ" :: Text)
+    ("C" :: Text)              --> ("ZZ" :: Text)
+    ("D" :: Text)              --> ("Or" :: Text)
+    ("E" :: Text)              --> ("Or" :: Text)
+    ("Or" :: Text)             --> ("A" :: Text)
+    ("F" :: Text)              --> ("B" :: Text)
+
+
+
```

Comment:
In general, please use *two* blank lines to separate top-level definitions. You should remove one here and at line 70, and add an extra one at line 65.